File: cpp/DIP/step1/RacingCar.cpp
class RacingCar {
public:
    RacingCar(int max_fuel) {
        max_fuel_ = max_fuel;
        remaining_fuel_ = max_fuel;
    }

    void Accelerate() {
        power_++;
        remaining_fuel_--;
    }

private:
    int max_fuel_;
    int remaining_fuel_;
    int power_;
};

File: python/DIP/step1/racing_car.py
class RacingCar:
    def __init__(self, max_fuel) -> None:
        super().__init__()
        self.__max_fuel = max_fuel
        self.__remaining_fuel = max_fuel
        self.__power = 0

    def accelerate(self):
        self.__power += 1
        self.__remaining_fuel -= 1
